src/ILVerify/tests/TestDataLoader.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace ILVerify.Tests
+{
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class TestDataLoader
+    {
+        /// <summary>
+        /// The folder with the binaries which are compiled from the test driver IL Code
+        /// </summary>
+        public static string TESTASSEMBLYPATH = @"..\..\..\ILTests\";
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Valid
+        /// The method must contain 1 '_'. The part before the '_' is a friendly name describing what the method does. 
+        /// The word after the '_' has to be 'Valid'. 
+        /// E.g.: 'SimpleAdd_Valid'
+        /// </summary>
+        public static TheoryData<TestCase> GetMethodsWithValidIL()
+        {
+            var methodSelector = new Func<string[], MethodDefinitionHandle, TestCase>((mparams, methodHandle) =>
+            {
+                if (mparams.Length == 2 && mparams[1].ToLower() == "valid")
+                {
+                    return new ValidILTestCase { MetaDataToken = MetadataTokens.GetToken(methodHandle) };
```

Comment:
Nit: The CLR codebase tends to spell the word *Metadata* as *MetaData* for reasons unknown, but metadata is a legit word found in the dictionaries since the seventies or so. I've been avoiding importing that capitalization plague into CoreRT. We don't spell things like PlaceHolder, NotePad, UnderLine with this weird capitalization either.